core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -442,54 +434,35 @@ protected void registerWebSocketEndpoints(ServerContainer container) {
 
   }
 
-  static void setUnsecurePathConstraints(
-      ServletContextHandler context,
-      List<String> unsecurePaths
-  ) {
-    //we need to set unsecure path only if there is an existing security handler. Otherwise all
-    // paths are by default unsecure
-    if (context.getSecurityHandler() != null && !unsecurePaths.isEmpty()) {
-      for (String path : unsecurePaths) {
-        Constraint constraint = new Constraint();
-        constraint.setAuthenticate(false);
-        ConstraintMapping constraintMapping = new ConstraintMapping();
-        constraintMapping.setConstraint(constraint);
-        constraintMapping.setMethod("*");
-        constraintMapping.setPathSpec(path);
-        ((ConstraintSecurityHandler) context.getSecurityHandler())
-            .addConstraintMapping(constraintMapping);
-      }
-    }
-  }
-
   static boolean enableBasicAuth(String authMethod) {
     return RestConfig.AUTHENTICATION_METHOD_BASIC.equals(authMethod);
   }
 
-  protected ConstraintSecurityHandler createBasicSecurityHandler(String realm, List<String> roles) {
+  protected ConstraintSecurityHandler createBasicSecurityHandler() {
+    final String realm = getConfiguration().getString(RestConfig.AUTHENTICATION_REALM_CONFIG);
+
     final ConstraintSecurityHandler securityHandler = new ConstraintSecurityHandler();
-    ConstraintMapping constraintMapping = createGlobalAuthConstraint(roles);
-    securityHandler.addConstraintMapping(constraintMapping);
+    securityHandler.addConstraintMapping(createGlobalAuthConstraint());
     securityHandler.setAuthenticator(new BasicAuthenticator());
     securityHandler.setLoginService(new JAASLoginService(realm));
     securityHandler.setIdentityService(new DefaultIdentityService());
     securityHandler.setRealmName(realm);
+
+    AuthUtil.createUnsecuredConstraints(config)
+        .forEach(securityHandler::addConstraintMapping);
+
     return securityHandler;
   }
 
+  @Deprecated // Use createGlobalAuthConstraint() or AuthUtil.createGlobalAuthConstraint(config)
   protected ConstraintMapping createGlobalAuthConstraint(List<String> roles) {
```

Comment:
We don't need to keep this method around, it's not a public api. We should just update any existing overrides to use `createGlobalAuthConstraint`

Reply:
OK